iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/VoltageAngleLimitAdderImpl.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.network.impl;
+
+import java.util.Optional;
+
+import com.powsybl.iidm.network.*;
+import com.powsybl.iidm.network.TerminalRef.Side;
+import com.powsybl.iidm.network.VoltageAngleLimit.FlowDirection;
+import com.powsybl.iidm.network.impl.util.Ref;
+
+/**
+ *
+ * @author Luma Zamarreño <zamarrenolm at aia.es>
+ * @author José Antonio Marqués <marquesja at aia.es>
+ */
+class VoltageAngleLimitAdderImpl implements VoltageAngleLimitAdder {
+
+    private final Ref<NetworkImpl> networkRef;
+    private TerminalRef from;
+    private TerminalRef to;
+    private double limit = Double.NaN;
+    private FlowDirection flowDirection = FlowDirection.BOTH_DIRECTIONS;
+
+    VoltageAngleLimitAdderImpl(Ref<NetworkImpl> networkRef) {
+        this.networkRef = networkRef;
+    }
+
+    @Override
+    public VoltageAngleLimitAdderImpl from(TerminalRef from) {
+        this.from = from;
+        return this;
+    }
+
+    @Override
+    public VoltageAngleLimitAdderImpl to(TerminalRef to) {
+        this.to = to;
+        return this;
+    }
+
+    @Override
+    public VoltageAngleLimitAdderImpl withLimit(double limit) {
+        this.limit = limit;
+        return this;
+    }
+
+    @Override
+    public VoltageAngleLimitAdderImpl withFlowDirection(FlowDirection flowDirection) {
+        this.flowDirection = flowDirection;
+        return this;
+    }
+
+    @Override
+    public VoltageAngleLimit add() {
+        if (limit <= 0) {
+            throw new IllegalStateException("Limit <= 0: " + Double.toString(limit));
+        }
+        Optional<Identifiable<?>> identifiableFrom = getIdentifiable(networkRef.get(), from);
+        if (identifiableFrom.isEmpty()) {
+            throw new IllegalStateException("Identifiable from not found: " + from.getId());
+        }
+        Optional<Identifiable<?>> identifiableTo = getIdentifiable(networkRef.get(), to);
+        if (identifiableTo.isEmpty()) {
+            throw new IllegalStateException("Identifiable to not found: " + to.getId());
+        }
+        Terminal terminalFrom = getTerminal(identifiableFrom.get(), from.getSide());
+        Switch switchFrom = getSwitch(identifiableFrom.get());
```

Comment:
No Switch at all as TerminalRef. Swicthes do not have any terminal, they cannot be used for terminal ref.

Reply:
Done. Switches are now not allowed as TerminalRef.